### PR TITLE
Reports: Orders

### PR DESCRIPTION
**Pull request by bordoni**
### 🎫 Ticket

[ET-1176]

### 🗒️ Description

When do you review this particular piece of code, keep in mind that the HTML for the Orders Page was not refreshed just yet, it is still using old code for the HTML and CSS.

### 🎥 Artifacts 
WIP

[ET-1176]: https://theeventscalendar.atlassian.net/browse/ET-1176

### CODE REVIEW

File: src/Tickets/Commerce/Reports/Orders.php [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+		$post = $attendees->get_post();
+
+		$commerce = tribe( Module::class );
+
+		$has_tickets = $commerce->get_attendees_count( $post->ID );
+		if ( ! $has_tickets ) {
+			return;
+		}
+
+		$tabbed_view = new \Tribe__Tickets__Commerce__PayPal__Orders__Tabbed_View();
```
Comment (juanfra) [inline src/Tickets/Commerce/Reports/Orders.php:204]: Shouldn't this orders report be agnostic and we should be hooking these from each gateway?

**Review by juanfra — COMMENTED** [review]
Looking really good! some comments around including PayPal references on the commerce orders generic functionality. I feel that if we don't trace the line now (when starting to implement) it can escalate to a situation where we have pieces of code related to gateways everywhere else.

File: src/Tickets/Commerce/Admin_Tables/Orders.php [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+	/**
+	 * Overrides the list of CSS classes for the WP_List_Table table tag.
+	 * This function is not hookable in core, so it needs to be overridden!
+	 *
+	 * @since TBD
+	 *
+	 * @return array List of CSS classes for the table tag.
+	 */
+	protected function get_table_classes() {
+		$classes = [ 'widefat', 'striped', 'tribe-tickets-commerce-report-orders' ];
```
[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tickets/Commerce/Admin_Tables/Orders.php:64]: I was going to leave that piece for you to handle at some point, most of this is copied from the Old code.

File: src/Tickets/Commerce/Admin_Tables/Orders.php [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+	/**
+	 * Orders Table constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$args = [
+			'singular' => 'order',
+			'plural'   => 'orders',
```
[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tickets/Commerce/Admin_Tables/Orders.php:48]: Not for internationalization it's for classes and slugs.

File: src/Tickets/Commerce/Reports/Orders.php [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $page_slug = 'tickets-commerce-orders';
+
+	/**
+	 * @var string
+	 */
+	public static $tab_slug = 'tribe-tickets-paypal-orders-report';
```
[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tickets/Commerce/Reports/Orders.php:35]: No can be removed.

File: src/Tickets/Commerce/Admin_Tables/Orders.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (juanfra) [reply src/Tickets/Commerce/Admin_Tables/Orders.php:48]: Cool I thought so! Thanks for the clarification :real_heart:

File: src/Tickets/Commerce/Reports/Report_Abstract.php [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	protected $template;
+
+	/**
+	 * Gets the template instance used to setup the rendering of the page.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Template
+	 */
+	public function get_template() {
```
[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tickets/Commerce/Reports/Report_Abstract.php:63]: The point of doing this is to avoid having the same template for everything.

**Pull request merged**